catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -110,14 +118,43 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
 
         SolrQuery query = getSolrQuery(request, filterDelegateFactory.newInstance(resolver));
 
+        boolean isFacetedQuery = false;
+        Serializable textFacetPropRaw = request.getPropertyValue(EXPERIMENTAL_FACET_PROPERTIES_KEY);
+
+        if (textFacetPropRaw != null && textFacetPropRaw instanceof FacetProperties) {
+            FacetProperties textFacetProp = (FacetProperties) textFacetPropRaw;
+            isFacetedQuery = true;
```

Comment:
Might consider adding a log statement here? Something really low like trace, and make sure to put a isTraceEnabled check on it so we don't performance suffer from it in prod

```java
if (LOGGER.isTraceEnabled()) {
    LOGGER.trace("Enabling faceted query for request [{}] on field {}", request, textFacetProp);
}

```